src/algebra/squarefree.lean

```diff
@@ -82,6 +82,20 @@ lemma squarefree_of_dvd_of_squarefree [comm_monoid R]
   squarefree x :=
 λ a h, hsq _ (h.trans hdvd)
 
+section squarefree_gcd_of_squarefree
+
+variables {α : Type*} [cancel_comm_monoid_with_zero α] [gcd_monoid α]
+
+lemma squarefree_gcd_of_squarefree_right (a : α) {b : α} (hb : squarefree b) :
+  squarefree (gcd a b) :=
+λ x hx, hb x $ hx.trans $ gcd_dvd_right _ _
```

Comment:
You may use [squarefree_of_dvd_of_squarefree](https://leanprover-community.github.io/mathlib_docs/algebra/squarefree.html#squarefree_of_dvd_of_squarefree) but the name is too long so the result will be longer ...

Reply:
Maybe `squarefree_of_dvd_of_squarefree` should be named to `squarefree.squarefree_of_dvd` or something like that (kind of like the lemmas `nat.coprime.coprime_dvd_left` and `nat.coprime.coprime_dvd_right`).

Reply:
Agreed, I'll make a quick PR to do so.

Reply:
Thanks, was planning to add it to this pr but forgot sorry